app/models/user_stats.rb

```diff
@@ -10,7 +10,39 @@ def self.exp_for(user)
     self.stats_for(user).exp
   end
 
+  def activity(date_range = nil)
+    {
+        exercises: {
+            solved_count: organization_exercises
+                              .joins(:assignments)
+                              .where(assignments: { top_submission_status: [:passed, :skipped], submitter: user }.merge_if_present(:submitted_at, date_range))
+                              .count,
+            count: organization_exercises.count},
+
+        messages: {
+            count: messages_in_discussions(date_range).count,
+            approved: messages_in_discussions(date_range).where(approved: true).count}
+    }
+  end
```

Comment:
The whole method looks like expensive in database terms. Not a big issue since this route will not be a bottleneck, but for example instead of querying `messages_in_discussions` twice, we could do it once and filter and count in-memory, or select both `count(approved)` and `count(1)`, or even group-by `approved`.

Reply:
Nice one. SQL is not my thing... 😅